src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -57,11 +71,30 @@ public Climber() {
     m_rightPIDController.setFF(0);
 
     Utilities.burnMotor(m_rightMotor);
+    m_leftLimitSwitch = m_leftMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+    m_rightLimitSwith = m_rightMotor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+    m_state = State.UNINITIALIZED;
+    initialize();
+  }
+
+  private void initialize() {
+    m_leftPIDController.setReference(ClimberConstants.kInitializingSpeed, ControlType.kVelocity);
+    m_rightPIDController.setReference(ClimberConstants.kInitializingSpeed, ControlType.kVelocity);
+  }
+
+  private double transformSpeed(double position, double speed) {
+    if (position >= ClimberConstants.kMaxExtendedLength - ClimberConstants.kMaxUnconstrainedPositionDifference) {
+      return (ClimberConstants.kMaxExtendedLength - position) * speed;
+    } else if (position >= ClimberConstants.kMinRetractedLength + ClimberConstants.kMaxUnconstrainedPositionDifference) {
+      return (ClimberConstants.kMinRetractedLength + position) * speed;
+    } else {
+      return speed;
+    }
   }
 
   public void startClimber(double leftSpeed, double rightSpeed) {
```

Comment:
This should be renamed to something corresponding to its function, e.g. `setSpeeds()`.